src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -934,8 +934,19 @@ public void clear() throws BackingStoreException {
         new SharedDatabasePreferences().clear();
     }
 
-    public void clear(String key) {
-        prefs.remove(key);
+    /**
+     * Clear all preferences.
+     *
+     * @throws IllegalArgumentException if the key does not exist
+     */
+    @Override
+    public void clear(String key) throws IllegalArgumentException {
```

Comment:
I would call the method `remove`, because it removes the key from the hashmap (instead of setting the value of the key to the empty string or to `null`).

```suggestion
    public void remove(String key) throws IllegalArgumentException {
```

Reply:
Not sure. What should be made public is that the preferences are cleared. How JabRefPreferences handles this internally should not be of concern to the outside...